### PR TITLE
Redir improve

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: libfreerdp/core/redirection.c [hunk 1]
```diff
@@ -373,14 +357,23 @@ static state_run_t rdp_recv_server_redirection_pdu(rdpRdp* rdp, wStream* s)
 		return STATE_RUN_FAILED;
 
 	Stream_Read_UINT16(s, flags);                  /* flags (2 bytes) */
+	if (flags != SEC_REDIRECTION_PKT)
+	{
+		char buffer1[1024] = { 0 };
+		char buffer2[1024] = { 0 };
+		WLog_ERR(TAG, "[%s] received invalid flags=%s, expected %s", __FUNCTION__,
+		         rdp_security_flag_string(flags, buffer1, sizeof(buffer1)),
+		         rdp_security_flag_string(SEC_REDIRECTION_PKT, buffer2, sizeof(buffer2)));
```
Comment (mfleisz) [inline libfreerdp/core/redirection.c:366]: Same as above - why not just use "SEC_REDIRECTION_PKT"

File: libfreerdp/core/capabilities.c [hunk 2]
```diff
@@ -4366,18 +4361,27 @@ BOOL rdp_recv_demand_active(rdpRdp* rdp, wStream* s)
...
+			         pdu_type_to_str(pduType, buffer2, sizeof(buffer2)));
+		}
+
+		// return FALSE;
 	}
 
 	rdp->settings->PduSource = pduSource;
 
-	if (!Stream_CheckAndLogRequiredLength(TAG, s, 8))
+	if (!Stream_CheckAndLogRequiredLength(TAG, s, 4))
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/core/capabilities.c:4378]: right, fixed.

File: libfreerdp/core/capabilities.c [hunk 3]
```diff
@@ -4366,18 +4361,27 @@ BOOL rdp_recv_demand_active(rdpRdp* rdp, wStream* s)
...
+			char buffer1[256] = { 0 };
+			char buffer2[256] = { 0 };
 
-		return FALSE;
+			WLog_ERR(TAG, "[%s] expected %s, got %s", __FUNCTION__,
+			         pdu_type_to_str(PDU_TYPE_DEMAND_ACTIVE, buffer1, sizeof(buffer1)),
+			         pdu_type_to_str(pduType, buffer2, sizeof(buffer2)));
+		}
+
+		// return FALSE;
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/core/capabilities.c:4373]: right, fixed.

File: libfreerdp/core/redirection.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/core/redirection.c:366]: `rdp_security_flag_string` does some nice formatting (e.g. stringify and add the number in brackets)

File: libfreerdp/core/capabilities.c [hunk 4]
```diff
@@ -4366,18 +4361,27 @@ BOOL rdp_recv_demand_active(rdpRdp* rdp, wStream* s)
...
 		if (pduType != PDU_TYPE_SERVER_REDIRECTION)
-			WLog_ERR(TAG, "expected %s %04x, got %04" PRIx16 "", pduName, PDU_TYPE_DEMAND_ACTIVE,
-			         pduType);
+		{
+			char buffer1[256] = { 0 };
+			char buffer2[256] = { 0 };
 
-		return FALSE;
+			WLog_ERR(TAG, "[%s] expected %s, got %s", __FUNCTION__,
+			         pdu_type_to_str(PDU_TYPE_DEMAND_ACTIVE, buffer1, sizeof(buffer1)),
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/core/capabilities.c:4369]: see below, formatting

**Review by mfleisz — APPROVED** [review]

**Pull request merged**